src/main/java/seedu/address/logic/parser/GradebookEditCommandParser.java

```diff
@@ -0,0 +1,84 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_GRADEBOOK_ITEM;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_GRADEBOOK_ITEM_EDIT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_GRADEBOOK_MAXMARKS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_GRADEBOOK_WEIGHTAGE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MODULE_CODE;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.GradebookEditCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.gradebook.Gradebook;
+
+/**
+ * Parses input arguments and creates a new GradebookEditCommand object
+ */
+public class GradebookEditCommandParser {
+    private static final String MESSAGE_MAX_MARKS_ERROR = "Invalid input. \nMaximum marks should only be an integer";
+    private static final String MESSAGE_WEIGHTAGE_ERROR = "Invalid input. \nWeightage should only be an integer";
+    private static final String MESSAGE_ERROR_EMPTY = "Module code and gradebook component name cannot be empty";
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the GradebookFindCommand
```

Comment:
@code String seems a little inappropriate as it uses the @code notation but String is just a class.

Reply:
AB4 has some javadoc comments like this too